web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -0,0 +1,341 @@
+import { Component, ElementRef, Input, OnChanges, SimpleChanges, ViewChild, AbstractType} from '@angular/core';
+import { formatDate } from '@angular/common'
+import * as d3 from 'd3';
+
+import { step189_2020 } from '../../../proto/step189_2020';
+
+/**
+ * BarChartData includes all data used by the bar chart.
+ */
+interface BarChartData {
+  pushID: string; // Push ID number
+  state: number;  // Tag of push end state
+  startTime: any; // Start time of push, in `YYYY-MM-DD HH:mm` format
+  duration: number; // Duration of push, in hour
+}
+
+type d3SVG = d3.Selection<SVGSVGElement, undefined, null, undefined>;
```

Comment:
Interesting that these are all different--maybe a comment to explain?

Reply:
Good point! I added some comments explaining them.